src/main/java/AudioPlayer.java

```diff
@@ -0,0 +1,261 @@
+     *
+     * @param title - this parameter is the title of the song to be searched by
+     * @return - this method return the singer of the song and it's number in the list.
+     */
+    public String searchSingerByTitle(String title) {
+        if (title != null && !title.isEmpty()) {
+            for (Song song : songs) {
+                if (song.getTitle().equalsIgnoreCase(title)) {
+                    StringBuilder songInfo = new StringBuilder("Singer: ");
+                    songInfo.append(song.getSingerName())
+                            .append("\nSong's number: ")
+                            .append(songs.indexOf(song) + 1);
+                    audioPlayerState.changeCurrentToPrevious();
+                    return songInfo.toString();
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * This method return list of songs in string format searched by the singer of the song.
+     *
+     * @param singer - this parameter is the singer of the song, whom we are looking for
+     * @return String - this method return list of songs in string format
+     */
+    public String searchSongsBySinger(Singer singer) {
+        StringBuilder songsBySinger = new StringBuilder();
+        songsBySinger.append("Songs by: ")
+                .append(singer.getName());
+
+        for (Song song : songs) {
+            if (song.checkSingerName(singer.getName())) {
+                songsBySinger.append(song);
+            }
+        }
+        audioPlayerState.changeCurrentToPrevious();
+        return songsBySinger.toString();
+    }
+
+    /**
+     * This method return number of songs that are present in the list
+     *
+     * @return - the returned value is the count of the songs in the list
+     */
+    public int size() {
+        audioPlayerState.changeCurrentToPrevious();
+        return songs.size();
+    }
+
+    /**
+     * This method adds a new song to the list
+     *
+     * @param song - this parameter is the actual song
+     */
+    public void add(Song song) {
+        if (song != null) {
+            songs.add(song);
+        }
+        audioPlayerState.changeCurrentToPrevious();
+    }
+
+    /**
+     * This method first check if the song is present in the list and then it's index is taken. After deletion current song index
+     * needs to be repositioned in case of deletion of the last song or after deletion of song, which was on smaller position in
+     * the list then currentSongIndex. In this cases currentSongIndex is repositioned one position backwards, so that currentSongIndex
+     * won't lose track of the current song.
+     *
+     * @param song - this parameter is the song to be deleted
+     */
+    public void remove(Song song) {
+        if (songs.contains(song)) {
+            int deleteSongIndex = songs.indexOf(song);
+            songs.remove(song);
+            if (currentSongIndex == songs.size() || deleteSongIndex < currentSongIndex)
```

Comment:
It's a valid structure, but please always add `{}` never mind there is only one line after the `if`

Reply:
+1